aries_cloudagent/protocols/connections/v1_0/manager.py

```diff
@@ -317,6 +335,14 @@ async def create_request(
 
         await connection.save(self.context, reason="Created connection request")
 
+        # MULTITENANCY: Add new key to handled keys to associate wallet to new connection.
+        ext_plugins = self.context.settings.get_value("external_plugins")
+        if ext_plugins and 'aries_cloudagent.wallet_handler' in ext_plugins:
+            wallet_handler: WalletHandler = await self.context.inject(WalletHandler)
+            wallet_id = self.context.settings.get_value("wallet.id")
+            wallet: BaseWallet = await self.context.inject(BaseWallet)
+            await wallet_handler.add_key(my_info.verkey, wallet_id)
+
```

Comment:
To be able to associate inbound messages for this connection a mapping between the `verkey` of this connection and the `wallet_id` is stored.